### PR TITLE
New control plane demo

### PR DESCRIPTION
**Pull request by tiagolobocastro**
Control plane demo for the node resource:
1. node service
2. rest service
3. node k8s operator

### CODE REVIEW

File: services/node/src/server.rs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+impl NodeStore {
+    pub async fn register(&self, registration: Register) {
+        let mut state = self.inner.state.lock().await;
+
+        let mut watchdog = Watchdog::new(self.inner.deadline);
+        let id = registration.id.clone();
+        let store = self.clone();
+        let deadline = self.inner.deadline;
+        watchdog.arm(async move {
```
Comment (gila) [inline services/node/src/server.rs:96]: just a small thing to consider is that as the number of nodes scales having a single timer that validates the deadline of all nodes might be useful?

File: services/node/src/server.rs [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    fn default() -> Self {
+        Self {
+            deadline: CliArgs::from_args().deadline.into(),
+            state: Default::default(),
+        }
+    }
+}
+
+impl NodeStore {
+    pub async fn register(&self, registration: Register) {
```
Comment (gila) [inline services/node/src/server.rs:89]: /// come comments if possible please..

File: rest/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+use actix_web::{get, web, App, HttpResponse, HttpServer, Responder};
+use mbus_api::*;
+use structopt::StructOpt;
+
+const REST_VERSION: &str = "v0";
```
Comment (gila) [inline rest/src/main.rs:5]: This version can be determined from the Toml file.

File: operators/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+[[bin]]
+name = "node-op"
+path = "node/src/main.rs"
+
+[dependencies]
+actix-web = "3.2.0"
+serde_json = "1.0"
+structopt = "0.3.15"
+log = "0.4.11"
+env_logger = "0.7"
```
Comment (gila) [inline operators/Cargo.toml:18]: I wonder if we should not use tracing here from the start? :thinking:

File: rest/src/main.rs (hunk 3 above)
Comment (tiagolobocastro) [reply rest/src/main.rs:5]: yeah good point. Not sure how we want to version the rest api tough, there seem to be 2 main ways of doing it, on the URI itself or through an http header.

**Review by gila — APPROVED** [review]

File: rest/src/main.rs (hunk 3 above)
Comment (gila) [reply rest/src/main.rs:5]: Yeah, good question, i *think* i prefer the version in the URI, but ... @jkryl  @mtzaurus what do you think?

File: services/node/src/server.rs (hunk 1 above)
Comment (tiagolobocastro) [reply services/node/src/server.rs:96]: yeah I did start on something that stored the last time a node was seen online (gave up for laziness) so we could make use of that on a single watchdog which checks for timeouts for all nodes based on the "last seen" time.

File: rest/src/main.rs (hunk 3 above)
Comment (bobek) [reply rest/src/main.rs:5]: :+1: for uri, it is simpler for developers to keep track of what they are calling. Especially when consumed by outside of the team.

File: operators/Cargo.toml (hunk 4 above)
Comment (tiagolobocastro) [reply operators/Cargo.toml:18]: yeah we probably should, I'll have a chat with @cjones1024

File: operators/node/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+
+    // Manage the MayastorNode CR
+    let nodes_api: Api<MayastorNode> =
+        Api::namespaced(kube_client.clone(), namespace);
+
+    loop {
+        // Poll for kubernetes nodes and rest nodes
+        // Reconcile from rest into kubernetes
+        if let Err(error) =
+            polling_work(&nodes_api, &rest_client, &rest_uri).await
```
Comment (AmitKumarDas) [inline operators/node/src/main.rs:85]: Do we ensure that k8s API server is not burdened with too many requests?

Comment (tiagolobocastro) [reply operators/node/src/main.rs:85]: No, it only sleeps 30s (or whatever we decide) between each poll.

**Review by jkryl — APPROVED** [review]

File: rest/src/main.rs (hunk 3 above)
Comment (mtzaurus) [reply rest/src/main.rs:5]: Having API version encoded in URL is more convenient, IMO.

**Review by gila — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors retry

**Comment by tiagolobocastro** [comment]
bors retry

**Pull request merged**